src/tasks.js

```diff
@@ -1,19 +1,167 @@
+          // edit task off selected id
+          li.classList.add('bg-col');
+          description.classList.add('desc');
+          Tasks.editTask(task.index);
+
+          // remove class edit
+          more.innerText = 'delete';
+          more.addEventListener('click', () => {
+            // delete selected item from dom
+            li.remove();
+            Tasks.deleteTask(task.index);
+            Tasks.setLocal();
+          });
+        });
+        ul.appendChild(li);
+      });
+      const li = document.createElement('li');
+      const p = document.createElement('p');
+      p.innerText = 'Clear all completed';
+      li.appendChild(p);
+      li.classList.add('d-flex', 'justify-content-around', 'bg', 'cursor');
+      li.addEventListener('click', () => {
+        Tasks.deleteCompleted();
+      });
+      ul.appendChild(li);
+    }
+
+    static getLocal() {
+      // get local storage
+      const tasks = JSON.parse(localStorage.getItem('tasks'));
+      if (tasks) {
+        Tasks.tasks = tasks;
+      }
+      Tasks.showTasks();
+    }
+
+    addTasks() {
+      // eslint-disable-next-line no-return-assign
+      Tasks.tasks.forEach((e, i) => e.index = i + 1);
+      // push from the last index deleted
+      Tasks.tasks.push(this);
+      // remove repeated tasks
+      Tasks.tasks = Tasks.tasks.filter((task, index, self) => index === self.findIndex((t) => (
+        t.description === task.description
+      )));
+      Tasks.setLocal();
+      // add Tasks to dom
+      Tasks.showTasks();
+    }
+
+    // delete task
+    static deleteTask(index) {
+      // delete from an array in order
+      if (index === 1 || this.tasks.length === 1) {
+        Tasks.tasks.splice(0, 1);
+      } else if (index === this.tasks.length) {
+        Tasks.tasks.splice(this.tasks.length - 1, 1);
+      } else {
+        Tasks.tasks.splice(index, 1);
+      }
+      // set local
+      Tasks.setLocal();
+    }
+
+    static editTask(index) {
+      // const li = document.querySelector(`li:nth-child(${index})`);
+      const description = document.querySelectorAll('p');
+      description.forEach((desc) => {
+        desc.setAttribute('contenteditable', 'true');
+        desc.addEventListener('blur', () => {
+          desc.setAttribute('contenteditable', 'false');
+          Tasks.tasks[index - 1].description = desc.innerText;
+          Tasks.setLocal();
+          Tasks.showTasks();
+        });
+        desc.addEventListener('keypress', (e) => {
+          if (e.key === 'Enter') {
+            desc.setAttribute('contenteditable', 'false');
+            Tasks.tasks[index - 1].description = desc.innerText;
+            Tasks.setLocal();
+            Tasks.showTasks();
+          }
+        });
+      });
+    }
+
+    // delete all tasks
+    static deleteAllTasks() {
+      Tasks.tasks = [];
+      Tasks.setLocal();
+    }
+
+    static deleteCompleted() {
+      const domStay = document.querySelectorAll('.completed');
```

Comment:
- [ ] **OPTIONAL**
The 'delete all completed' button is within the `<ul>` tag and this is semantically not right, don't you think it'd be better if we moved the button out of the tasks list because it's not a task right? 😄 Please tell me what you think!

Reply:
It looks nice, but this is not me selecting the whole ul, this is me selecting the whole list that are completed

Reply:
However i implemented your suggestions, and it looks better